src/games/coinflip/index.ts

```diff
@@ -0,0 +1,223 @@
+import { SuiClient, SuiEvent } from "@mysten/sui.js/client";
+import {
+    TransactionBlock as TransactionBlockType,
+    TransactionObjectArgument,
+    TransactionResult
+} from "@mysten/sui.js/transactions";
+
+import { randomBytes } from 'crypto-browserify';
+
+import { 
+  BLS_VERIFIER_OBJ, 
+  COIN_MODULE_NAME,
+  LIMBO_CORE_PACKAGE_ID,
+  LIMBO_MAX_MULTIPLIER, 
+  LIMBO_MIN_MULTIPLIER, 
+  LIMBO_MODULE_NAME, 
+  LIMBO_PACKAGE_ID, 
+  UNIHOUSE_PACKAGE, 
+  UNI_HOUSE_OBJ
+} from "../../constants";
+import { extractGenericTypes, sleep } from "../../utils";
+
+export interface CoinFlipInput {
+    betType: 0 | 1;
+    coinPackageId: string;
+    coinType: string;
+    stakeCoin: TransactionObjectArgument;
+    transactionBlock: TransactionBlockType;
+}
+
+ export interface CoinFlipGameIdInput {
+    coinPackageId: string;
+    coinType: string;
+    pollInterval: number;
+    transactionResult: any;
+}
+
+interface InternalCoinFlipGameIdInput extends CoinFlipGameIdInput {
+    suiClient: SuiClient;
+}
+
+export interface WeightedCoinFlipInput {
+    coinType: string;
+    multiplier: number;
+    stakeCoin: TransactionObjectArgument;
+    transactionBlock: TransactionBlockType;
+}
+
+export interface WeightedCoinFlipGameIdInput {
+    coinType: string;
+    pollInterval: number;
+    transactionResult: any;
+}
+
+interface InternalWeightedCoinFlipGameIdInput extends WeightedCoinFlipGameIdInput {
+    suiClient: SuiClient;
+}
+
+// Add coinflip to the transaction block
+export const createCoinflip = ({
+    transactionBlock,
```

Comment:
@healthyred changed `txb` here to `transactionBlock` to match the mysten convention in their sdk. i thought it'd be a little better since that's what developers will be used to seeing.

we can change this if it will break anything though.